source/index.html.md

```diff
@@ -18,91 +18,121 @@ search: true
 
 # Authentication
+To manage the data via our API your application needs to gain access on behalf of the user. This is done through obtaining an access token via [OAuth2](https://tools.ietf.org/html/rfc6749). The access token must then be send in each request in the HTTP header like this: "Authorization: Bearer TOKEN".
 
-You can start playing with the Kontist API using [HTTP Basic Auth](https://en.wikipedia.org/wiki/Basic_access_authentication). However this is not to be used in production as it is highly rate limited. The documentation of our oAuth flow is in preparation and will be added at a later point in time.
+You need two kinds of credentials to get such a token: The first part is a fixed pair of client id and client secret. They identify your client application which connects to the API. Each application has its own pair of client id and secret, please contact us via developer@kontist.com to get your own client credentials.
+The second part is obtained through the user and can be done in several ways, here we describe the preferred way through the "Authorization Code" grant type.
 
-`Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l`
+## Authorization Code
+In general, the process looks like this:
+1. You redirect the user in a browser to an url on our end.
+2. The user is required to login and needs to accept your applications authorization request. The browser redirects back to your application with a `code` parameter.
+3. Your application can then exchange this `code` together with the `client_secret` into an `access_token` through a backend request to our API.
 
-<aside class="notice">
-You must replace <code>QWxhZGRpbjpPcGVuU2VzYW1l</code> with a base64-encoded representation of <code>your_username:your_password</code>.
-</aside>
-
-## JWT auth
+Let us go through the process step by step. At first we need to send the user to a special url in the browser:
+```http
+https://api.kontist.com/api/oauth/authorize?scope=offline&response_type=code&client_id=78b5c170-a600-4193-978c-e6cb3018dba9&redirect_uri=https://your-application/callback&state=OPAQUE_VALUE
+```
 
-You can read more about JWT token auth under [https://jwt.io/introduction](https://jwt.io/introduction).
+Adjust the parameters like this:
+| Parameter | Description |
+| --------- | ----------- |
+| scope | space delimited list of scopes your application is going to access. Please see the list below.|
+| response_type | Set fixed as "code". |
+| client_id | This is your client id you got from us. Do not include the secret here.|
+| redirect_uri | This is your applications callback url which is bound to your client id.|
+| state | Can be used to verify our response. You can put in anything here and we will send it back to your application. |later.
```

Comment:
what's with this `later.` at the end after closing the table? 😄is it on purpose?

Reply:
Done. ;-)